### ISSUE TITLE
NvEncSharp.Sample.VideoDecode shows only blue channel

### ISSUE THREAD

**Issue by vlt-arizona** [jlennox/NvEncSharp#6]
Following your earlier comment about this test code, I converted my video to raw h264 like so:
ffmpeg -i input.mp4 -c copy output.h264
The file output.h264 displays the converted video but I only see the blue channel. Is there any special setup involved?

**Comment by jlennox** [comment]
No, that is correct.

If you try opening the file using ffplay or perhaps vlc (I believe it supports raw h264?) does it decode properly?

**Comment by vlt-arizona** [comment]
No, it’s not the display code. The “Save as bitmap” images are also blue.

Vaughn

 

From: ***@***.*** ***@***.***> 
Sent: Thursday, October 21, 2021 8:28 AM
To: 'jlennox/NvEncSharp' ***@***.***>
Subject: RE: [jlennox/NvEncSharp] NvEncSharp.Sample.VideoDecode shows only blue channel (Issue #6)

 

Thank you for getting back to me so quickly!

Yes, it decodes properly in both ffplay and vlc.

FFprobe gives the following stats:

 

Input #0, h264, from 'Big_Buck_Bunny_1080_10s_1MB.h264': 0B f=0/0

  Duration: N/A, bitrate: N/A

  Stream #0:0: Video: h264 (High), yuv420p(progressive), 1920x1080 [SAR 1:1 DAR 16:9], 30 fps, 30 tbr, 1200k tbn

    nan M-V:    nan fd=   0 aq=    0KB vq=   92KB sq=    0B f=0/0

 

I’ve tried this with a number of other converted videos; they all show blue only. I also created a version of this video with saturation=0 to verify that the phenomenon had nothing to do with the colors of the original. Just now I tried converting the mp4 video with a different version of ffmpeg; the result is the same. Maybe it could be the display code?

Vaughn

**Comment by vlt-arizona** [comment]
OK, trying again. 
![output-00000000-1](https://user-images.githubusercontent.com/89151762/138358766-ecf09248-a33f-4540-bd40-d9b08fdcc936.jpg)
![output-00000000-2](https://user-images.githubusercontent.com/89151762/138358793-e616227f-2fd4-43a0-9dbb-195120ca142a.jpg)
n.
